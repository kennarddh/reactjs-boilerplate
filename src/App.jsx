import React from 'react'

import styled from 'styled-components'

export const Button = styled.button`
	color: red;
`

const App = () => {
	return (
		<div>
			<div data-testid='custom-element'>heading</div>
			<Button data-testid='button'>Test</Button>
		</div>
	)
}

export default App
