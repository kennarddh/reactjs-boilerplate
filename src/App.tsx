import { FC } from 'react'

import Button from 'Components/Button'

const App: FC = () => {
	return (
		<div>
			<div data-testid='custom-element'>heading</div>
			<Button>Blue Button</Button>
			<Button red>Red Button</Button>
		</div>
	)
}

export default App
