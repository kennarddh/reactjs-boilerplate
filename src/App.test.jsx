import React from 'react'

import { render, screen } from '@testing-library/react'

import App, { Button } from '@/App'

describe('app', () => {
	it('should have text heading', () => {
		expect.assertions(1)

		render(<App />)

		expect(screen.getByTestId('custom-element')).toHaveTextContent(
			'heading'
		)
	})

	it('should render button', () => {
		expect.assertions(1)

		const { asFragment } = render(<Button />)

		expect(asFragment()).toMatchSnapshot()
	})
})
