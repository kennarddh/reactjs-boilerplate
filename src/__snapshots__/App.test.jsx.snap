// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render button 1`] = `
<DocumentFragment>
  .c0 {
  color: red;
}

<button
    class="c0"
  />
</DocumentFragment>
`;
